aodndata/gsla/handler.py

```diff
@@ -48,14 +48,24 @@ def get_gsla_type(filepath):
         return fields['product_type']
 
     elif GSLA_REGEX_YEARLY.match(file_basename):
-        return os.path.join('DM00', 'yearfiles')
+        return os.path.join(get_product_type(filepath), 'yearfiles')
 
     else:
         raise InvalidFileNameError(
             "file name: \"{filename}\" not matching regex to deduce dest_path".format(
                 filename=file_basename))
 
 
+def get_product_type(netcdf_path):
+    with Dataset(netcdf_path, mode='r') as nc_obj:
+        if hasattr(nc_obj, 'product_type'):
```

Comment:
Can just invert this (EAFP, https://en.wikipedia.org/wiki/Python_syntax_and_semantics#Exceptions) to get rid of the redundant hasattr:

```python
try:
    return nc_obj.product_type
except AttributeError:
    raise InvalidInputFileError(
                "Expecting 'product_type' attribute in netCDF'{gzip}'".format(gzip=os.path.basename(netcdf_path))
            )
```